### PR TITLE
Outdated packages number on output

### LINKED ISSUE #20: Outdated packages number on output
Before:

```
Some packages are not up to date:
```

After:

```
1 package is not up to date:
```

OR

```
42 packages are not up to date:
```

### PR DESCRIPTION
**Pull request by EmanueleMinotto**
Closes #20

### CODE REVIEW

File: src/VersionsCheck.php [hunk 1]
```diff
@@ -101,7 +101,11 @@ public function versionCompare($version1, $operator, $version2)
 
     private function createNotUpToDateOutput(array &$output)
     {
-        $output[] = '<warning>Some packages are not up to date:</warning>';
+        $output[] = sprintf(
+            '<warning>%d %s not up to date:</warning>',
+            count($this->outdatedPackages),
+            1 != count($this->outdatedPackages) ? 'packages are' : 'package is'
```
Comment (Soullivaneuh) [inline src/VersionsCheck.php]: Using `count` twice will cost useless resources IMO.

Would be better to store the count on a result variable before like `$outdatedPackagesCount`.

File: tests/VersionsCheckTest.php [hunk 2]
```diff
@@ -125,14 +125,11 @@ public function testMultiplePackagesComparison(array $packagesData, $preferStabl
...
-        $this->assertSame(sprintf(<<<EOF
-<warning>Some packages are not up to date:</warning>
-
-%s
-
-
-EOF
-        , implode("\n", $shouldBeUpdatedOutput)), $this->versionsCheck->getOutput());
+        $expected = sprintf(
+            "/^<warning>[0-9]+ package(s are| is) not up to date:<\/warning>\n\n%s\n\n$/",
```
Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: The exact expected number would be more reliable.

Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: And BTW, I would like to keep `<<<EOF` method.

**Comment by Soullivaneuh** [comment]
Thanks for your contribution. :-)

After review, please squash your commit on remove issue reference on commit comment.

File: tests/VersionsCheckTest.php [hunk 3]
```diff
@@ -99,8 +99,9 @@ public function getOutdatedDetectionTestData()
      *
      * @param array $packagesData
      * @param bool  $preferStable
+     * @param int   $outdatedPackages
      */
-    public function testMultiplePackagesComparison(array $packagesData, $preferStable = false)
+    public function testMultiplePackagesComparison(array $packagesData, $preferStable = false, $outdatedPackages)
```
Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: `$outdatedPackagesCount`

File: tests/VersionsCheckTest.php [hunk 4]
```diff
@@ -125,14 +126,15 @@ public function testMultiplePackagesComparison(array $packagesData, $preferStabl
 
         $this->checkPackages();
 
-        $this->assertSame(sprintf(<<<EOF
-<warning>Some packages are not up to date:</warning>
+        $expectedContent = str_replace('/', '\\/', quotemeta(implode("\n", $shouldBeUpdatedOutput)));
+        $this->assertRegExp(sprintf(<<<EOF
+/^<warning>%d packages are not up to date:<\/warning>
```
Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: Why keeping regexp for this? Single string comparison with a `sprintf` would be enough.

File: src/VersionsCheck.php [hunk 5]
```diff
@@ -101,7 +101,13 @@ public function versionCompare($version1, $operator, $version2)
 
     private function createNotUpToDateOutput(array &$output)
     {
-        $output[] = '<warning>Some packages are not up to date:</warning>';
+        $totalOutdatedPackages = count($this->outdatedPackages);
```
Comment (Soullivaneuh) [inline src/VersionsCheck.php]: * `$outdatedPackagesCount`
* I think the newline can be removed, it's for the same logic. :+1:

**Comment by Soullivaneuh** [comment]
Tests are failing, could you please take a look?

Note: I suggest you to ping me when it's done. Maintainers don't get notification for just commit on PR.

File: tests/VersionsCheckTest.php [hunk 6]
```diff
@@ -126,13 +127,12 @@ public function testMultiplePackagesComparison(array $packagesData, $preferStabl
...
         $this->assertSame(sprintf(<<<EOF
-<warning>Some packages are not up to date:</warning>
+<warning>%d packages are not up to date:</warning>
 
 %s
 
-
 EOF
-        , implode("\n", $shouldBeUpdatedOutput)), $this->versionsCheck->getOutput());
+            , $outdatedPackagesCount, implode("\n", $shouldBeUpdatedOutput)), $this->versionsCheck->getOutput());
```
Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: Please keep indent.

File: tests/VersionsCheckTest.php [hunk 7]
```diff
@@ -126,13 +127,12 @@ public function testMultiplePackagesComparison(array $packagesData, $preferStabl
         $this->checkPackages();
 
         $this->assertSame(sprintf(<<<EOF
-<warning>Some packages are not up to date:</warning>
+<warning>%d packages are not up to date:</warning>
 
 %s
 
-
```
Comment (Soullivaneuh) [inline tests/VersionsCheckTest.php]: Should not be removed. This is why tests are failing IMO.

**Comment by EmanueleMinotto** [comment]
@Soullivaneuh should be ok now, sorry for the repeated mistakes ^^'

**Comment by Soullivaneuh** [comment]
Thank you. :+1: 

Will merge it later.

**Comment by Soullivaneuh** [comment]
@EmanueleMinotto Available on [v1.1.0](https://github.com/Soullivaneuh/composer-versions-check/releases/tag/v1.1.0).

Sorry for the delay.

**Pull request merged**